### DISCUSSION TITLE
[BUG] Add-PnPListItemComment - format-default: Exception has been thrown by the target of an invocation.

### DISCUSSION THREAD

**Discussion by ehrenschlueter** [pnp/powershell#3855] [General] [answered]
### Reporting an Issue or Missing Feature
Connected to Sharepoint and other PnP commands work normally.

Add-PnPListItemComment -List 'Demo List' -Identity 9 -Text "Hello world"
format-default: Exception has been thrown by the target of an invocation.

### Expected behavior 
Post comment to Modern SharePoint List Item.

### Actual behavior 
The comment is posted to the SharePoint list and terminates with an error message of: 
"format-default: Exception has been thrown by the target of an invocation."

When this command is in a loop, it kills the script and does not honor any sort of error handling to bypass the termination to override it, too.

![image](https://github.com/pnp/powershell/assets/15827921/23febfb9-6cc2-471f-8ec3-d1d63d8a4b6f)

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)

$connection = Connect-PnPOnline -Url "https://yourtenant.sharepoint.com/sites/DemoSite" -ClientId <your client id> -Tenant yourtenant.onmicrosoft.com -CertificateBase64Encoded $yourcert

#throws an error response and posts the message to the list item.
Add-PnPListItemComment -List 'Demo List' -Identity 9 -Text "Hello world"

#This command will terminate the script / loop in its tracks instead of silently continuing
Add-PnPListItemComment -List 'Demo List' -Identity 9 -Text "Hello world" -ErrorAction SilentlyContinue -WarningAction SilentlyContinue

### What is the version of the Cmdlet module you are running?
2.4.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows

### PowerShell Version 
7.4.1

**Comment by ehrenschlueter** [comment]
@gautamdsheth Why was this "bug" converted to a discussion?

I am working on a workaround via the Graph api for my personal code however I'm not really sure how it would help this project due to that it's in C# and my code is in PowerShell. :-)

**Comment by ehrenschlueter** [comment, marked as answer]
@gautamdsheth Code away! :-D

My workaround to the situation is to leverage Power Automate / Logic App and target the API directly.

I successfully reproduced the issue by using this content type:
"content-type": "application/json;charset=UTF-8"

The issue was resolved after using this content type:
"content-type": "application/json;odata=verbose"

**::Working::** 
params:
{
  "dataset": "https://yourtenant.sharepoint.com/sites/DemoSite",
  "parameters/method": "POST",
  "parameters/uri": "_api/web/lists/GetByTitle('Demo List')/items(9)/Comments",
  "parameters/headers": {
    "accept": "application/json;odata=verbose",
    "content-type": "application/json;odata=verbose"
  },
  "parameters/body": "{\n   \"__metadata\": {\n     \"type\": \"Microsoft.SharePoint.Comments.comment\"\n   },\n  \"text\": \"Hello World\"\n}\n"
}


Applying the same knowledge learned from the above to the PnP module and using "Invoke-PnPSPRestMethod" the following works:

```powershell
Function Add-PnPListItemComments {
	Param (
		$List,
		$Identity,
		$Text
	)

 	$metadata = @{
		"type" = "Microsoft.SharePoint.Comments.comment"
	}

	$Content = @{
		"__metadata" = $metadata
		"text" = $Text
	} 

	$ContentType = "application/json;odata=verbose"

 	Invoke-PnPSPRestMethod -Method Post -Url "/_api/web/lists/GetByTitle('$List')/items($Identity)/Comments()" -Content $Content -ContentType $ContentType
}

Add-PnPListItemComments -List 'Demo List' -Identity '9' -Text 'Hello World' 
```

**Reply by SabbeRubbish** [reply]
Thank you aquaus for this great workaround - worked like a charm. Hoping there will be a fix soon.

**Reply by HensHub** [reply]
Worked great @aquaus!

Any idea how I can force the new entries createdDate an author property?

I tried to add them like this but this doesn't seem to work :(

```
	$Content = @{
		"__metadata" = $metadata
		"author" = $Author
		"createdDate" = $CreatedDate
		"text" = $Text
	} 
```

**Reply by ehrenschlueter** [reply]
I think it's a limitation on the Microsoft.SharePoint.Comments.comment endpoint type with Sharepoint Online. Please open a case with Microsoft SharePoint team and let me know what you find out. When I was originally researching the fix for this; I didn't need to post the comments as the user and instead using a Managed Identity was perfectly okay. :-)

**Reply by HensHub** [reply]
@aquaus, createdDate and author are in the endpoint but they don't seem to accept any data, or might be overwritten immediately by Sharepoint. I solved it now by just making a comment where the text has the user included. Why I need this is cause I try to copy a list from one site to another including attachments version history and comments. I kind of got everything working but the comments seem to be not that accessible. Copying a list from one site to another with all its contents and so on should in my opinion already be a standard Sharepoint feature. But anyway I got more or less what I want. Thanks anyway :)
